drivers/source/MinimalSerial.cpp

```diff
@@ -0,0 +1,130 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2019 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "drivers/MinimalSerial.h"
+
+#if DEVICE_SERIAL
+
+namespace mbed {
+
+MinimalSerial::MinimalSerial(
+    PinName tx,
+    PinName rx,
+    int baud = MBED_CONF_PLATFORM_DEFAULT_SERIAL_BAUD_RATE
+) : _serial(), _baud(baud)
+{
+    _serial_ptr = &_serial;
+
+    init(tx, rx);
+}
+
+MinimalSerial::MinimalSerial(
+    serial_t *serial_ptr,
+    bool serial_inited,
+    PinName tx,
+    PinName rx,
+    int baud = MBED_CONF_PLATFORM_DEFAULT_SERIAL_BAUD_RATE
+) : _serial(), _baud(baud)
+{
+    if (serial_inited) {
+        return;
+    }
+    _serial_ptr = serial_ptr;
```

Comment:
It does not matter at the moment since we are looking at code size but `_serial_ptr` should be assigned before the if statement above. In fact, it should be assigned in the initializatio list.